### PR TITLE
Add EventRepository

### PR DESCRIPTION
**Pull request by antooni**

### CODE REVIEW

File: packages/backend/src/peripherals/database/migrations/020_events.ts [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+If you find that something was incorrectly set up in the `up` function you
+should create a new migration file that fixes the issue.
+
+*/
+
+import { Knex } from 'knex'
+
+export async function up(knex: Knex) {
+  await knex.schema.createTable('events', function (table) {
```
Comment (sz-piotr) [inline packages/backend/src/peripherals/database/migrations/020_events.ts:17]: Add indexes for `unix_timestamp`, `project_id` and `time_span`

File: packages/backend/src/peripherals/database/EventRepository.ts [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+    this.getAll = this.wrapGet(this.getAll)
+    this.deleteAll = this.wrapDelete(this.deleteAll)
+    this.getByProjectAndName = this.wrapGet(this.getByProjectAndName)
+
+    /* eslint-enable @typescript-eslint/unbound-method */
+  }
+
+  async getByProjectAndName(
+    projectId: ProjectId,
+    name: string,
```
Comment (sz-piotr) [inline packages/backend/src/peripherals/database/EventRepository.ts:31]: We probably would like to allow specifying multiple names at once

**Review by sz-piotr — CHANGES_REQUESTED** [review]

File: packages/backend/src/peripherals/database/migrations/020_events.ts (hunk 1 above)
Comment (antooni) [reply packages/backend/src/peripherals/database/migrations/020_events.ts:17]: The index created in private key works for single columns, I tested it by sql analyze

File: packages/backend/src/peripherals/database/EventRepository.ts (hunk 2 above)
Comment (antooni) [reply packages/backend/src/peripherals/database/EventRepository.ts:31]: Do you mean event names, or both event names and projectId?

Comment (sz-piotr) [reply packages/backend/src/peripherals/database/EventRepository.ts:31]: event names

**Pull request closed without merging**